OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -63,6 +63,8 @@ public class ArmamentInfo : ConditionalTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		public readonly bool ShootAtClosestTargetablePosition = false;
```

Comment:
> We'll want this in TS too right?

Yes.

> TargetAcquisition { CenterPosition, Footprint } 

I'd like to point out that I'm planning to later use this to make weapons shoot at custom TargetPositions on non-buildings as well, so this won't really be limited to `Footprint`s.